### ISSUE TITLE
Incorrect list of symbols in PHP

### ISSUE THREAD

**Issue by 50Wliu** [atom/symbols-view#175]
_From @sebastien-gallet on January 17, 2016 2:40_

In PHP, if a function name first character is lowercased and the parameter is a strongly typed initialized array:
![screen shot 2016-01-17 at 10 59 52 am](https://cloud.githubusercontent.com/assets/968164/12375536/62d04934-bd0a-11e5-8be5-c0fafefcfaf4.png)

Then pressing the shortcut "Goto Symbol", the list of symbols shows the parameter name instead of the function name.
(As a side note, somehow it pick "name" in the comment as a symbol as well.) 
![screen shot 2016-01-17 at 11 00 00 am](https://cloud.githubusercontent.com/assets/968164/12375537/62d39d96-bd0a-11e5-8c15-6dc8ad518a15.png)

_Copied from original issue: atom/language-php#118_

**Comment by 50Wliu** [comment]
_From @tommyvdv on January 25, 2016 14:55_

I believe i'm having the same issue in this example and hope a few extra examples can help you pinpoint the source of the problem.

<img width="221" alt="1" src="https://cloud.githubusercontent.com/assets/83648/12554291/03d475f4-c37c-11e5-98ce-6e282573a888.png">
<img width="244" alt="2" src="https://cloud.githubusercontent.com/assets/83648/12554290/03c53b84-c37c-11e5-96c0-13398ecebb49.png">
<img width="242" alt="3" src="https://cloud.githubusercontent.com/assets/83648/12554289/03addebc-c37c-11e5-99b2-776791968c01.png">

The symbol definition for an optional parameter named `$locale` seems to override the method name.

In dense classes i'm also getting some hits on the code inside methods. This makes it hard to navigate some files. Disregard this if this behavior is intentional (I'm new to Atom and was a sublime user).

<img width="371" alt="4" src="https://cloud.githubusercontent.com/assets/83648/12554288/028c0dce-c37c-11e5-920d-94d65bab17d6.png">

> edit: I've discovered that the package `goto` solves the inaccurate symbols.

**Comment by 50Wliu** [comment]
_From @bj7 on May 7, 2016 17:46_

I also experience an issue with the symbol list not being correct in PHP files. I have a function:
![image](https://cloud.githubusercontent.com/assets/1395096/15093554/ca055db4-1459-11e6-9924-274bba83471e.png)
When using Command-R (Mac) and searching on `view`, all I get back is the name of my class:
![image](https://cloud.githubusercontent.com/assets/1395096/15093561/ef4e53e6-1459-11e6-82e8-3a0c70c7a045.png)

Which file handles symbol list definitions? I would be glad to investigate.

**Comment by hultberg** [comment]
I'm seeing this issue all the time. I also struggle to find several methods in big classes. I end up just searching for them with `cmd+f`

**Comment by bj7** [comment]
I'll try to take a look at this in my spare time to determine a cause.

**Comment by mcdado** [comment]
Moreover, declaration of class properties are reported as symbols, just like local variables inside class methods!

<img width="756" alt="screen shot 2016-08-19 at 21 21 02" src="https://cloud.githubusercontent.com/assets/898057/17821633/eb095092-6652-11e6-9655-d0c3024bd5ce.png">

This makes symbols-view in php almost useless.

**Comment by nathanpitman** [comment]
I've had the same issue, as someone switching from Sublime the behaviour of Cmd-R is not what I expected and pretty much useless. The Goto package is perfect though. :)

**Comment by godcong** [comment]
happy new year issus

**Comment by pablopunk** [comment]
@50Wliu any plans on this issue? It's a big deal when I use atom at work

**Comment by 50Wliu** [comment]
Nope.  Anyone is free to investigate this issue and submit a PR though.
